python/kvikio/kvikio/utils.py

```diff
@@ -41,6 +40,8 @@ def __init__(
         root_path: str | pathlib.Path,
         range_support: bool = True,
         max_lifetime: int = 120,
+        handler: type[BaseHTTPRequestHandler] | None = None,
```

Comment:
Not sure how much it matters, but the changes to the public API here *should* be backwards compatible.

We just need a way to inject a custom HTTP handler here, so that we can control when a 500 is returned.